api/controllers/loomies.go

```diff
@@ -182,3 +184,78 @@ func HandleValidateLoomieExists(c *gin.Context) {
 		"loomie_id": loomie_id,
 	})
 }
+
+func HandleCaptureLoomie(c *gin.Context) {
+	loomie_req := interfaces.LoomieForm{}
+	userid, _ := c.Get("userid")
+
+	if err := c.BindJSON(&loomie_req); err != nil {
+		c.AbortWithStatusJSON(http.StatusBadRequest, gin.H{"error": true, "message": "Bad request"})
+		return
+	}
+
+	if loomie_req.LoomieId == "" {
+		c.AbortWithStatusJSON(http.StatusBadRequest, gin.H{"error": true, "message": "No Loomie ID was provided"})
+		return
+	}
+
+	zoneRadiusStr := configuration.GetEnvironmentVariable("GAME_ZONE_RADIUS")
+	zoneRadius, _ := strconv.ParseFloat(zoneRadiusStr, 64)
+	loomie, err := models.ValidateLoomieExists(loomie_req.LoomieId)
+
+	if err != nil {
+		if err == mongo.ErrNoDocuments {
+			c.AbortWithStatusJSON(http.StatusNotFound, gin.H{"error": true, "message": "Loomie was not found"})
+			return
+		} else {
+			c.AbortWithStatusJSON(http.StatusInternalServerError, gin.H{"error": true, "message": "Internal server error"})
+			return
+		}
+	}
+
+	if math.Abs(loomie.Latitude-loomie_req.Latitude) > zoneRadius || math.Abs(loomie.Longitude-loomie_req.Longitude) > zoneRadius {
```

Comment:
You can use the `useNear` function to avoid repeating code. Check the `/api/útiles/helpers.go` file.